roles/ceph-update/tasks/main.yml

```diff
@@ -8,69 +8,68 @@
   delay: 60
   delegate_to: "{{ groups['ceph_monitors'][0] }}"
 
-- name: stop ceph mon
-  service: name=ceph-mon-all state=stopped
-  when: "'ceph_monitors' in group_names"
+# tasks of ceph monitors
+- block:
+  - name: stop ceph mon
+    service: name=ceph-mon-all state=stopped
+  
+  - name: correct ceph dir owner
+    file: path=/var/lib/ceph state=directory owner=ceph group=ceph recurse=yes
+    when: upgrade_ceph2jewel|bool
 
-- name: correct ceph dir owner
-  file: path=/var/lib/ceph state=directory owner=ceph group=ceph recurse=yes
-  when: upgrade_ceph2jewel|bool and "ceph_monitors" in group_names
+  - name: correct owner of admin keyring
+    file: path=/etc/ceph/ceph.client.admin.keyring owner=ceph group=ceph
+    when: upgrade_ceph2jewel|bool
```

Comment:
The `file` module is idempotent, I think we should remove the conditional here and just enforce the ownership we want.

Honestly, I think it would be better to enforce that ownership in the location(s) that the file gets created rather than later here.